### PR TITLE
docs(Configuration): document new devServer.overlay options

### LINKED ISSUE #6808: Document dev server overlay improvement
**Feature to document**

There are new dev server overlay configurations with the following changes:

- https://github.com/webpack/webpack-dev-server/pull/4605
- https://github.com/webpack/webpack-dev-server/pull/4813

**Author(s)**

@malcolm-kee

**Additional information**

- [X] I am willing to work on this issue and submit a pull request.

I plan to work on this next week, but happy to let others work on this if they're interested.

### PR DESCRIPTION
**Pull request by malcolm-kee**
- [x] Read and sign the [CLA][1]. PRs that haven't signed it won't be accepted.
- [x] Make sure your PR complies with the [writer's guide][2].
- [x] Review the diff carefully as sometimes this can reveal issues.
- [x] Do not abandon your Pull Request: [Stale Pull Requests][3].

Resolves 

- #6808 

[1]: https://github.com/openjs-foundation/EasyCLA#openjs-foundation-cla
[2]: https://webpack.js.org/contribute/writers-guide/
[3]: https://webpack.js.org/contribute/#pull-requests

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ Once you are in the project directory, run the following commands:
 - `yarn build` to create a production version of the site.
 - `yarn start` to develop on a local webpack-dev-server: [localhost:3000][3].
 
-> NOTE: run `yarn fetch` before running `yarn start` command for the first time
+> NOTE: run `yarn fetch-repos` and then `yarn fetch` before running `yarn start` command for the first time
```
Comment (malcolm-kee) [inline .github/CONTRIBUTING.md:22]: I not sure if the better solution is to rename `fetch-repos` script to `fetch:repos` or update this docs, but definitely this is something that is outdated and blocking first time setup.

File: src/content/configuration/dev-server.mdx [hunk 2]
```diff
@@ -299,9 +311,35 @@ module.exports = {
 Usage via the CLI:
 
 ```bash
-npx webpack serve --client-overlay-errors --no-client-overlay-warnings
+npx webpack serve --client-overlay-errors --no-client-overlay-warnings --client-overlay-runtime-errors
+```
+
+To filter based on the thrown error, you can pass a function that accepts an `error` parameter and returns boolean.
```
Comment (snitin315) [inline src/content/configuration/dev-server.mdx:317]:
```suggestion
To filter based on the thrown error, you can pass a function that accepts an `error` parameter and returns a boolean.
```

File: src/content/configuration/dev-server.mdx [hunk 3]
```diff
@@ -299,9 +311,35 @@ module.exports = {
...
+
+For example, to ignore error thrown by [`AbortController.abort()`](https://developer.mozilla.org/en-US/docs/Web/API/AbortController/abort):
```
Comment (snitin315) [inline src/content/configuration/dev-server.mdx:319]:
```suggestion
For example, to ignore an error thrown by [`AbortController.abort()`](https://developer.mozilla.org/en-US/docs/Web/API/AbortController/abort):
```

File: src/content/configuration/dev-server.mdx [hunk 4]
```diff
@@ -299,9 +311,35 @@ module.exports = {
...
+          }
+          return true;
+        },
+      },
+    },
+  },
+};
 ```
 
+W> The function will not have access to variables declared in the outer scope within the configuration file.
```
Comment (snitin315) [inline src/content/configuration/dev-server.mdx:341]:
```suggestion
W> The function will not have access to the variables declared in the outer scope within the configuration file.
```

**Review by snitin315 — COMMENTED** [review]
Thanks for documenting this! A few suggestions and then LGTM

File: src/content/configuration/dev-server.mdx (hunk 3 above)
Comment (malcolm-kee) [reply src/content/configuration/dev-server.mdx]: I change to "errors".

**Review by snitin315 — APPROVED** [review]
LGTM. Thank you!

**Comment by malcolm-kee** [comment]
@snitin315  Who should I tag to merge this?

**Review by TheLarkInn — APPROVED** [review]
Thank you for the contribution! 🎉

**Comment by TheLarkInn** [comment]
@chenxsan whats this authorization check I'm happy to force merge but unsure if this is okay?

**Comment by snitin315** [comment]
We need to ask @sokra to invite more team members to vercel so that we can also autherize deploy previews.

**Comment by snitin315** [comment]
This is good to merge.

**Comment by chenxsan** [comment]
See
https://github.com/webpack/webpack.js.org/pull/6776#issuecomment-1531404049

**Pull request merged**